### PR TITLE
Handle coordinate transforms with infinite rect.width()

### LINKED ISSUE #47904: 3.24.x/3.22.5 has exception "vector too long" on Windows 10
### What is the bug or the crash?

I have a project with a large number of relatively complex layers, mostly PostgreSQL, but also some files and others, along with some tiled base maps.  The project worked fine in 3.22.4, but has an uncaught exception in 3.24, 3.24.1, and 3.22.5 which pops up a dialog of "vector too long".  This happens just after the project loads, and after dismissing the exception dialog a couple of times (maybe related to the number of layers with issues?), I can pan and zoom and some other things just fine - I haven't tried anything fancy.  However, if I try to exit, even if I discard changes to the project, it pops up again and stops qgis-bin.exe from exiting.  The only solution is to kill the task.  This doesn't happen with other projects, even a simplified version of the project with issues.  However, I've not been able to figure out which layers cause the problem.  When I saw the bug in 3.24 I was hoping to find the time to bisect through removing layers, but I haven't yet, but I wanted to report this now that I see it has not been fixed in 3.24.1 and has turned up in 3.22.5.  Maybe that particular exception is obvious to someone, especially happening in the exit code path.

### Steps to reproduce the issue

I don't have a good reproducible test case.  I'll try to install debugging symbols and see if I can catch the exception in a debugger.

### Versions

3.24.0, 3.24.1, 3.22.5

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeremylea**
Try to handle invalid rectangles better, which seems to be fallout from #34518.  Also, catch any type of exception in the return path to be more robust (otherwise it is impossible to close the application).

Fixes #47904.

### CODE REVIEW

**Comment by nyalldawson** [comment]
@jeremylea can you please split this into two seperate PRs? I'd like to discuss/review the changes individually

**Comment by jeremylea** [comment]
I don't know how to do that.

**Comment by jeremylea** [comment]
Sorry, should have tagged @nyalldawson.  I'm not good with git or Github, I use it because I have to.

**Comment by Oxalin** [comment]
Was it ever pulled in? After @nyalldawson request to split the commit as two separated PR and @jeremylea stating that he didn't know how to do it, the code seems to have been rotting.

File: src/core/proj/qgscoordinatetransform.cpp [hunk 1]
```diff
@@ -538,16 +540,28 @@ QgsRectangle QgsCoordinateTransform::transformBoundingBox( const QgsRectangle &r
         throw QgsCsException( QObject::tr( "Could not transform bounding box to target CRS" ) );
       yMax = 90 - EPS;
     }
+    if (xMin < -180)
+    {
+      if (xMax < -180)
+        throw QgsCsException(QObject::tr("Could not transform bounding box to target CRS"));
```
Comment (Oxalin) [inline src/core/proj/qgscoordinatetransform.cpp:546]: The exception could be more talkative by letting know the error and pinpointing it. But the same comment could go for any of the exceptions around this code.

**Pull request closed without merging**